web3/datastructures.py

```diff
@@ -54,23 +67,23 @@ def _repr_pretty_(self, builder, cycle):
         builder.text(")")
 
     @classmethod
-    def _apply_if_mapping(cls, value):
+    def _apply_if_mapping(cls, value: Any) -> Any:
```

Comment:
This might be better typed as:

```python
T = TypeVar("T")
TValue = TypeVar("TValue")

@classmethod
def _apply_if_mapping(cls: Type[T], value: TValue) -> Union[T, TValue]:
    ...
```